azure-iot-device/azure/iot/device/common/errors.py

```diff
@@ -0,0 +1,151 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
>Microsoft [](start = 16, length = 9)

open issue:  This contains a set of "base" errors.  Some are truly common (like ConnectionFailed), and others are IoTHub specific (like BadDeviceResponseError),  and even more are not truly common, but they are common between iothub and provisioning (like ThrottlingError).  From what I've seen in other SDKs, nobody has really taken the time to get a good look at what errors can be returned by what services, and that will eventually have to happen, but I'm not sure if that's a windmill that I want to tilt at.

Reply:
I agree, for now, I'm just happy to have errors, we can reorganize a bit later.

---
In reply to: [302267292](https://github.com/Azure/azure-iot-sdk-python-preview/pull/136#discussion_r302267292) [](ancestors = 302267292)